### PR TITLE
Fix flaky ADS unit tests by using DOM rendering

### LINKED ISSUE #23509: Investigate flaky unit tests failing intermittently after vscode source refresh
There are a handful of unit tests failing intermittently in the canary builds that need to be investigated.  I'll use this issue to keep a list of these tests and provide status on fixing, disabling, etc.

Initial list of tests to investigate:

```
TerminalLinkManager
    getLinks and open recent link
      ✔ should return web links in order
      1) should return web links in order

DefaultConfiguration
    ✔ configuration default overrides are read from environment
    1) configuration default overrides are read from environment

Getting Started Markdown Renderer
    ✔ renders theme picker markdown with images
    1) renders theme picker markdown with images

A couple other ones...
1) Buffer Content Tracker
       should cap the size of the cached lines, removing old lines in favor of new lines:

 2) Buffer Content Tracker
       should cap the size of the cached lines, removing old lines in favor of new lines:
```

### PR DESCRIPTION
**Pull request by lewis-sanchez**
This PR fixes #23509

This PR fixes an issue with our flaky unit tests that are caused because DOM rendering isn't being used. The fix is temporary and will most likely need to be removed once GPU accelerating is working again in xterm. This PR also reenables all tests that were skipped previously.

### CODE REVIEW

File: src/vs/platform/configuration/test/common/testConfigurationService.ts [hunk 1]
```diff
@@ -18,6 +18,21 @@ export class TestConfigurationService implements IConfigurationService {
 	readonly onDidChangeConfiguration = this.onDidChangeConfigurationEmitter.event;
 
 	constructor(configuration?: any) {
+		// {{SQL CAROB EDIT}} - START
+		// Ensures that all configuration services use the DOM renderer.
```
Comment (kisantia) [inline src/vs/platform/configuration/test/common/testConfigurationService.ts:22]: can you add a little more detail here on why this `SQL CARBON EDIT` is needed in the comments?

File: src/vs/platform/configuration/test/common/testConfigurationService.ts [hunk 2]
```diff
@@ -18,6 +18,23 @@ export class TestConfigurationService implements IConfigurationService {
 	readonly onDidChangeConfiguration = this.onDidChangeConfigurationEmitter.event;
 
 	constructor(configuration?: any) {
+		// {{SQL CARBON EDIT}} - START
```
[the remark this answers is not in the document]
Comment (lewis-sanchez) [reply src/vs/platform/configuration/test/common/testConfigurationService.ts:21]: @Charles-Gagnon, do you know if VS Code ran into a similar issue? I don't know all the details around the conversation with Daniel from VS Code.

**Review by kisantia — APPROVED** [review]

File: src/vs/platform/configuration/test/common/testConfigurationService.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Charles-Gagnon) [reply src/vs/platform/configuration/test/common/testConfigurationService.ts:21]: I don't believe so, or at least I've never seen them having any issues in their tests and Daniel didn't say anything about fixing it on their side.

My best guess was that it was something with the container they're running their tests in - they have their own custom container and many of the scripts they run are different from ours so something there might be our 

I did try updating to the latest xterm packages but that didn't solve it either unfortunately.

**Review by Charles-Gagnon — APPROVED** [review]
Thanks for taking care of this @lewis-sanchez! Fingers crossed it helps....

**Pull request merged**